analytics/analytics.go

```diff
@@ -183,8 +186,8 @@ func (dc *dataCollection) sendData() {
 	req.Header.Add("Content-Type", "application/json")
 
 	res, err := http.DefaultClient.Do(req)
+	defer res.Body.Close()
```

Comment:
Still not right - it must be *after* the `if err != nil` otherwise it will throw the nil pointer because res is nil, the example I had was:
```
client := http.DefaultClient
resp, err := client.Do(req)
if err != nil {
    return nil, err
}
defer resp.Body.Close()
```